src/main/kotlin/org/rust/lang/core/resolve/ref/RsLitExprReferenceContributor.kt

```diff
@@ -62,3 +74,60 @@ private class RsLiteralFileReferenceSet(
         }
     }
 }
+
+private val pathValueLiteral: PsiElementPattern.Capture<RsLitExpr> = psiElement<RsLitExpr>()
+    .with("onPathLiteral") { expr -> expr.kind is RsLiteralKind.String && isPathLitExpr(expr) }
+
+private fun isPathLitExpr(expr: RsLitExpr): Boolean {
+    val (function, arguments) = getFunctionAndArguments(expr) ?: return false
+    val owner = ((function.owner as? RsAbstractableOwner.Impl)?.impl?.typeReference?.type as? TyAdt)?.item
+    val (implLookup, knownItems) = expr.implLookupAndKnownItems
+    val isCallExpr = arguments.parent is RsCallExpr
+
+    return when {
+        // Path::new(<literal>)
+        owner == knownItems.Path && function.name == "new" -> true
+        // PathBuf::from(<literal>)
+        owner == knownItems.PathBuf && function.name == "from" -> true
+        // fn foo<P: AsRef<Path>>(p: P) -> foo(<literal>)
+        // or fn foo(p: impl AsRef<Path>) -> foo(<literal>)
+        else -> {
+            var argumentIndex = arguments.exprList.indexOf(expr)
+            if (argumentIndex == -1) return false
+
+            // UFCS, skip self argument
+            if (function.hasSelfParameters && isCallExpr && argumentIndex > 0) {
+                argumentIndex -= 1
+            }
+
+            val parameter = function.valueParameters.getOrNull(argumentIndex) ?: return false
+            isAsRefPathGeneric(implLookup, knownItems, parameter) || isImplAsRefPath(knownItems, parameter)
+        }
+    }
+}
+
+// fn foo<P: AsRef<Path>>(p: P)
+private fun isAsRefPathGeneric(lookup: ImplLookup, knownItems: KnownItems, parameter: RsValueParameter): Boolean {
+    val type = (parameter.typeReference?.type as? TyTypeParameter) ?: return false
+    return lookup.getEnvBoundTransitivelyFor(type).any { isAsRefPath(knownItems, it) }
+}
+
+// fn foo(p: impl AsRef<Path>)
+private fun isImplAsRefPath(knownItems: KnownItems, parameter: RsValueParameter): Boolean {
+    val type = (parameter.typeReference?.type as? TyAnon) ?: return false
+    return type.traits.any { isAsRefPath(knownItems, it) }
+}
+
+private fun isAsRefPath(knownItems: KnownItems, trait: BoundElement<RsTraitItem>): Boolean =
+    trait.element == knownItems.AsRef && (trait.positionalTypeArguments.getOrNull(0) as? TyAdt)?.item == knownItems.Path
+
+private fun getFunctionAndArguments(expr: RsLitExpr): Pair<RsFunction, RsValueArgumentList>? {
+    expr.parentOfType<RsCallExpr>()?.let {
+        val path = (it.expr as? RsPathExpr)?.path ?: return null
+        return Pair(path.reference?.resolve() as? RsFunction ?: return null, it.valueArgumentList)
+    }
+    expr.parentOfType<RsMethodCall>()?.let {
```

Comment:
And after such changes, I suppose it will be possible to simplify this method.
For example something like:
```kotlin
return when (val grandParent = expr.context?.context) {
    is RsCallExpr -> { /*do something*/ }
    is RsMethodCall -> { /*do something*/ }
    else -> null
}
```